modules/gapi/src/api/gcomputation.cpp

```diff
@@ -201,19 +213,23 @@ cv::GRunArgs cv::GComputation::apply(GRunArgs &&ins, GCompileArgs &&args)
             }
             case cv::GShape::GARRAY:
             {
-                switch (info.kind)
-                {
-                    case cv::detail::OpaqueKind::CV_POINT2F:
-                        run_args.emplace_back(cv::detail::VectorRef{std::vector<cv::Point2f>{}});
-                        outs.emplace_back(cv::util::get<cv::detail::VectorRef>(run_args.back()));
-                        break;
-                    default:
-                        util::throw_error(std::logic_error("Unsupported kind for GArray"));
-                }
+                cv::detail::VectorRef ref;
+                util::get<cv::detail::ConstructVec>(info.ctor)(ref);
```

Comment:
Not sure if any other function in this file/class operates on the same _low_ level of abstraction - it is all built on some higher-level entities. Please keep the abstraction level.